### PR TITLE
sql-parser: parse CREATE SECRET

### PR DESCRIPTION
**Pull request by mkysel**
Add support for CREATE SECRET required for the platform. For the motivation see #11022.

### CODE REVIEW

File: src/sql/src/plan/statement/ddl.rs [hunk 1]
```diff
@@ -2354,6 +2372,7 @@ pub fn plan_drop_objects(
         | ObjectType::Sink
         | ObjectType::Type => plan_drop_items(scx, object_type, if_exists, names, cascade),
         ObjectType::Role => plan_drop_role(scx, if_exists, names),
+        ObjectType::Secret => unreachable!("DROP SECRET"),
```
Comment (benesch) [inline src/sql/src/plan/statement/ddl.rs:2375]: You'll want `bail_unreachable!` here, I think, to avoid panicking!

**Review by benesch — APPROVED** [review]
LGTM!

File: src/sql/src/plan/statement/ddl.rs (hunk 1 above)
Comment (mkysel) [reply src/sql/src/plan/statement/ddl.rs:2375]: `bail_unsupported`?

Comment (mkysel) [reply src/sql/src/plan/statement/ddl.rs:2375]: I will open a follow up PR for DROP SECRET and make this change

Comment (benesch) [reply src/sql/src/plan/statement/ddl.rs:2375]: Sorry, yeah, `bail_unsupported`.

**Pull request merged**